doc/dev/adapters.md

```diff
@@ -1,15 +1,699 @@
+
+func (builder) SetListEntryTypes(map[string]*listentry.Type){
+  ...
+}
+
+
+////////////////// Runtime Methods //////////////////////////
+
+var checkResult = adapter.CheckResult{
+  Status:        rpc.Status{Code: int32(rpc.OK)},
+  ValidDuration: 1000000000 * time.Second,
+  ValidUseCount: 1000000000,
+}
+
+func (handler) HandleListEntry(context.Context, *listentry.Instance) (adapter.CheckResult, error) {
+  return checkResult, nil
+}
+
+func (handler) Close() error { return nil }
+
+////////////////// Bootstrap //////////////////////////
+
+// GetInfo returns the Info associated with this adapter implementation.
+func GetInfo() adapter.BuilderInfo {
+  return adapter.BuilderInfo{
+     Name:        "istio.io/mixer/adapter/noop2",
+     Description: "Does nothing",
+     SupportedTemplates: []string{
+        listentry.TemplateName,
+     },
+     NewBuilder: func() adapter.HandlerBuilder { return builder{} },
+     DefaultConfig:        &types.Empty{},
+  }
+}
+
+```
+* Sample of a no-op adapter that supports the above sample 'quota' Template.
+
+```
+type (
+  builder struct{}
+  handler struct{}
+)
+
+// ensure our types implement the requisite interfaces
+var _ quota.HandlerBuilder = builder{}
+var _ quota.Handler = handler{}
+
+///////////////// Configuration Methods ///////////////
+
+func (builder) Build(Context.Context, adapter.Env) (adapter.Handler, error) {
+  return handler{}, nil
+}
+func (builder) SetAdapterConfig(adapter.Config)                      {}
+func (builder) Validate() (*adapter.ConfigErrors)                 { return }
+
+func (builder) SetQuotaTypes(map[string]*quota.Type){
+  ...
+}
+
+////////////////// Runtime Methods //////////////////////////
+
+func (handler) HandleQuota(ctx context.Context, _ *quota.Instance, args adapter.QuotaRequestArgs) (adapter.QuotaResult2, error) {
+  return adapter.QuotaResult2{
+        ValidDuration: 1000000000 * time.Second,
+        Amount:        args.QuotaAmount,
+     },
+     nil
+}
+
+func (handler) Close() error { return nil }
+
+////////////////// Bootstrap //////////////////////////
+
+// GetInfo returns the Info associated with this adapter implementation.
+func GetInfo() adapter.BuilderInfo {
+  return adapter.BuilderInfo{
+     Name:        "istio.io/mixer/adapter/noop2",
+     Description: "Does nothing",
+     SupportedTemplates: []string{
+        quota.TemplateName,
+     },
+     NewBuilder: func() adapter.HandlerBuilder { return builder{} },
+     DefaultConfig:        &types.Empty{},
+  }
+}
+```
+
+
+The above section provided a complete example of a simple adapter. In the next sections we'll look in more detail about
+how to build mixer with custom adapters that are are not shipped with the default Mixer build, and step-by-step guide to
+build a simple adapter.
+
+# Summary diagrams
+
+Below diagrams show the relationship between, Adapter, Mixer, Template and operator config. It also shows the flow of
```

Comment:
Below diagrams -> The diagrams below
between adapters, templates, Mixer, and operator config.

Reply:
It also shows -> They also show

Reply:
D